yarpl/include/yarpl/single/SingleOperator.h

```diff
@@ -46,30 +47,84 @@ class SingleOperator : public Single<D> {
     }
 
     void observerOnSuccess(D value) {
-      observer_->onSuccess(std::move(value));
-      upstreamSubscription_.reset(); // should break the cycle to this
+      terminateImpl(TerminateState::Down(), folly::Try<D>{std::move(value)});
+    }
+
+    void observerOnError(folly::exception_wrapper ew) {
+      terminateImpl(TerminateState::Down(), folly::Try<D>{std::move(ew)});
     }
 
     Reference<Operator> getOperator() {
       return single_;
     }
 
-   private:
+    void terminateErr(folly::exception_wrapper ew) {
+      terminateImpl(TerminateState::Both(), std::move(ew));
+    }
+
+    // SingleSubscription.
+
+    void cancel() override {
+      folly::exception_wrapper ew{std::runtime_error{"Cancelled"}};
```

Comment:
why to create the exception_wrapper when you call temrinate(Up()) which only cancels the subscription but never onErrors?

Reply:
Hmm yeah, good point.  I think I did that because I needed to pass *something* to the `folly::Try<D>` argument.

I can pass an empty `Try` instead.